src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/MixinPathNavigationRegion.java

```diff
@@ -0,0 +1,152 @@
+import io.github.opencubicchunks.cubicchunks.utils.Coords;
+import io.github.opencubicchunks.cubicchunks.world.CubeCollisionGetter;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.SectionPos;
+import net.minecraft.util.Mth;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.PathNavigationRegion;
+import net.minecraft.world.level.chunk.ChunkAccess;
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(PathNavigationRegion.class)
+public abstract class MixinPathNavigationRegion implements CubeCollisionGetter {
+
+    @Mutable @Shadow @Final protected int centerX;
+    @Mutable @Shadow @Final protected int centerZ;
+    @Shadow protected boolean allEmpty;
+    @Shadow @Final protected Level level;
+    private int centerY;
+    private int diameter;
+    private boolean isCubic;
+    private int minY;
+    private int height;
+
+    private ChunkAccess[] cubes;
+
+//    @Redirect(method = "<init>", at = @At(value = "NEW", target = "net/minecraft/world/level/chunk/ChunkAccess"))
+//    private ChunkAccess[][] no2DArray(Level level, BlockPos blockPos, BlockPos blockPos2) {
+//        if (!((CubicLevelHeightAccessor) level).isCubic()) {
+//            return new ChunkAccess[][]
+//        }
+//    }
+
+
+    @Redirect(method = "<init>", at = @At(value = "INVOKE", target = "Lnet/minecraft/core/SectionPos;blockToSectionCoord(I)I"))
+    private int zeroOutExistingIterators(int coord, Level world, BlockPos blockPos, BlockPos blockPos2) {
+        if (!((CubicLevelHeightAccessor) world).isCubic()) {
+            return SectionPos.blockToSectionCoord(coord);
+        }
+
+        return 0;
+    }
+
+
+    @Inject(method = "<init>", at = @At("TAIL"))
+    private void useCubes(Level world, BlockPos blockPos, BlockPos blockPos2, CallbackInfo ci) {
+        this.isCubic = ((CubicLevelHeightAccessor) world).isCubic();
+        this.minY = world.getMinBuildHeight();
+        this.height = world.getHeight();
+
+        if (!isCubic) {
+            return;
+        }
+        this.centerX = Coords.blockToCube(blockPos.getX());
+        this.centerY = Coords.blockToCube(blockPos.getY());
+        this.centerZ = Coords.blockToCube(blockPos.getZ());
+
+        int offsetX = Coords.blockToCube(blockPos2.getX());
+        int offsetY = Coords.blockToCube(blockPos2.getY());
+        int offsetZ = Coords.blockToCube(blockPos2.getZ());
```

Comment:
Are these variables really supposed be called `center` and `offset`? It looks like these are the corners of a cuboid, not its center and radii.

Reply:
Mmmm maybe